File: PHP/643. Maximum Average Subarray I.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $nums
     */
    public function findMaxAverage(array $nums, int $k): float
    {
        $left = 0;
        $average = PHP_INT_MIN;
        $sum = 0;

        for ($right = 0; $right < count($nums); $right++) {
            $window = $right - $left + 1;

            $sum += $nums[$right];

            if ($window === $k) {
                $average = max($sum / $k, $average);

                $sum -= $nums[$left];
                $left++;
            }
        }

        return $average;
    }
}

File: Python/643. Maximum Average Subarray I.py
import math
from typing import List


# noinspection PyMethodMayBeStatic, PyPep8Naming
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left, right = 0, 0
        average = -math.inf
        summation = 0

        while right < len(nums):
            window = right - left + 1

            summation += nums[right]

            if window == k:
                average = max(summation / k, average)

                summation -= nums[left]
                left += 1

            right += 1

        return average
